### PR TITLE
[exporter/awsemfexporter] Add app signals specific user agent.

### PR DESCRIPTION
**Pull request by jefchien**
**Description:** Adds additional user agent if configured for App Signals. Extracts the SDK Telemetry attributes from the resource attributes to build the user agent strings. Uses a TTL cache to keep track of the language/version pairs received by the exporter. 

**Testing:** Built (w/ AWS SDK LogLevel set to `LogDebug`).
```
│ -----------------------------------------------------                                                                                                                                                                                                                                                                                                                  │
│ 2024/02/26 23:05:41 DEBUG: Request logs/PutLogEvents Details:                                                                                                                                                                                                                                                                                                          │
│ ---[ REQUEST POST-SIGN ]-----------------------------                                                                                                                                                                                                                                                                                                                  │
│ POST / HTTP/1.1                                                                                                                                                                                                                                                                                                                                                        │
│ Host: logs.us-east-2.amazonaws.com                                                                                                                                                                                                                                                                                                                                     │
│ User-Agent: CWAgent/Unknown (go1.20.12; linux; amd64) ID/2cad80d1-f569-451e-86af-6ca4668c0e84 inputs:(otlp) processors:(awsappsignals resourcedetection) outputs:(app_signals awsemf awsxray cloudwatchlogs) CWAgent/Unknown (AppSignals) aws-sdk-go/1.48.6 (go1.20.12; linux; amd64) telemetry-sdk-java/1.31.0                                                        │
```

### CODE REVIEW

File: .chloggen-aws/sdk-telemetry-user-agent.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
```
Comment (lisguo) [inline .chloggen-aws/sdk-telemetry-user-agent.yaml:1]: naive q: should we be making a chloggen entry whenever we contribute to our contrib repo?

File: exporter/awsemfexporter/internal/appsignals/useragent.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+// Process takes the telemetry SDK language and version and adds them to the cache. If it already exists in the cache
+// and has the same value, extends the TTL. If not, then it sets it and rebuilds the user agent strings.
+func (h *UserAgent) Process(labels map[string]string) {
+	language := labels[semconv.AttributeTelemetrySDKLanguage]
+	version := labels[semconv.AttributeTelemetrySDKVersion]
+	if language != "" && version != "" {
+		value := h.cache.Get(language)
+		if value == nil || value.Value() != version {
```
Comment (lisguo) [inline exporter/awsemfexporter/internal/appsignals/useragent.go:69]: value is the language right? wouldn't it always be != to version? Shouldn't we check if languageValue != language or versionValue != version?

File: exporter/awsemfexporter/internal/appsignals/useragent.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+// formatStr formats the telemetry SDK language and version into the user agent format.
+func formatStr(language, version string) string {
+	return fmt.Sprintf("telemetry-sdk-%s/%s", language, version)
+}
+
+// Process takes the telemetry SDK language and version and adds them to the cache. If it already exists in the cache
+// and has the same value, extends the TTL. If not, then it sets it and rebuilds the user agent strings.
+func (h *UserAgent) Process(labels map[string]string) {
+	language := labels[semconv.AttributeTelemetrySDKLanguage]
```
Comment (lisguo) [inline exporter/awsemfexporter/internal/appsignals/useragent.go:65]: should we consider some validation on the values of these attributes? Maybe at least have a limitation of x character length?

File: exporter/awsemfexporter/internal/appsignals/useragent.go (hunk 2 above)
Comment (jefchien) [reply exporter/awsemfexporter/internal/appsignals/useragent.go:69]: `value` is the version. It's a map of language-version pairs.

File: .chloggen-aws/sdk-telemetry-user-agent.yaml (hunk 1 above)
Comment (jefchien) [reply .chloggen-aws/sdk-telemetry-user-agent.yaml:1]: We decided to do it to keep track of our changes, but it isn't enforced well.

File: exporter/awsemfexporter/internal/appsignals/useragent.go (hunk 3 above)
Comment (jefchien) [reply exporter/awsemfexporter/internal/appsignals/useragent.go:65]: Yep can have a length limit.

**Review by lisguo — APPROVED** [review]

**Review by movence — APPROVED** [review]

**Review by lisguo — APPROVED** [review]

**Review by movence — APPROVED** [review]

**Pull request merged**